02-household-queries/chainlit-household-bot.py

```diff
@@ -137,18 +173,73 @@ async def set_llm_model():
     await msg.stream_token(f"Done setting up {llm_name} LLM")
     await msg.send()
 
+async def set_embeddings():
+    settings = cl.user_session.get("settings")
+    embeddings = settings["embedding"]
+    msg = cl.Message(
+        author="backend",
+        content=f"Setting up embedding: `{embeddings}`...\n",
+    )
+    embedding = None
+    if embeddings in GOOGLE_EMBEDDINGS:
+        GOOGLE_API_KEY = os.environ.get('GOOGLE_API_KEY')
+        model_name= embeddings.split('::')[1]
+        embedding =  GoogleGenerativeAIEmbeddings(model=model_name, google_api_key=GOOGLE_API_KEY)
+    elif embeddings in OPEN_SOURCE_EMBEDDINGS:
+        embedding = SentenceTransformerEmbeddings(model_name="all-MiniLM-L6-v2")
+    elif embeddings in HUGGING_FACE_EMBEDDINGS:
+        model_name= embeddings.split('::')[1]
+        embeddings = HuggingFaceEmbeddings(model_name=model_name)
+    else:
+        await cl.Message(content=f"Could not initialize embedding: {embeddings}").send()
+        return
+    cl.user_session.set("embedding", embedding)
+    await msg.stream_token(f"Done setting up {embeddings} embedding")
+    await msg.send()
+
+async def set_vector_db():
+    await init_embedding_function_if_needed()
+    embeddings = cl.user_session.get("embedding")
+    msg = cl.Message(
+        author="backend",
+        content=f"Setting up Chroma DB with `{embeddings}`...\n",
+    )
+    
+    # clean up db when setting embedding for embedding dimension does not match collection dimensionality
+    persistent_client = chromadb.PersistentClient(settings=Settings(allow_reset=True), path="./chroma_db")
+    cl.user_session.set("chromadb_client", persistent_client)
```

Comment:
`chromadb_client` isn't used anywhere else. Should this line be:
```suggestion
    cl.user_session.set("persistent_client", persistent_client)
```